dvc/render/vega_converter.py

```diff
@@ -161,23 +177,108 @@ def convert(
         data: Dict,
         revision: Optional[str] = None,
         filename: Optional[str] = None,
+        skip: List = None,
+        config_path: str = "",
     ):
         """
         Convert the data. Fill necessary fields ('x', 'y') and return both
         generated datapoints and updated properties.
         """
+        if not skip:
+            skip = []
+
         processed = deepcopy(data)
 
-        for _, step in self.steps:
-            processed = step(processed)  # type: ignore
+        for step_name, step in self.steps:
+            if step_name not in skip:
+                processed = step(  # type: ignore
+                    processed,
+                    revision=revision,
+                    filename=filename,
+                    config_path=config_path,
+                )
 
         self._infer_y(processed)  # type: ignore
 
         if revision:
-            for datapoint in processed:
-                datapoint[REVISION_FIELD] = revision
+            # TODO what to do with revision and filename -
+            #      how to detect if it was already provided?
+            if REVISION_FIELD not in first(processed).keys():
+                for datapoint in processed:
+                    datapoint[REVISION_FIELD] = revision
         if filename:
             for datapoint in processed:
                 datapoint[FILENAME_FIELD] = filename
 
         return processed, {**self.props, **self.inferred_props}
+
+    def _generate_y_values(  # noqa: C901 # TODO
+        self,
+        datapoints: List[Dict],
+        y_values: Optional[Union[str, List, Dict]],
+        revision: str,
+        filename: str,
+        config_path: str,
+        **kwargs,
+    ) -> List[Dict]:
+        # TODO, after tests
+        if not y_values:
+            return datapoints
+
+        result = []
+        properties_update = {}
+
+        def _generate_y(datapoint, field):
+            tmp = datapoint.copy()
+
+            tmp["dvc_inferred_y_value"] = datapoint[field]
+            tmp[REVISION_FIELD] = f"{revision}::{filename}::{field}"
+
+            if (
+                "y_label" not in properties_update
+                and "y_label" not in self.props
+            ):
+                properties_update["y_label"] = "y"
+
+            properties_update["y"] = "dvc_inferred_y_value"
+
+            return tmp
+
+        if isinstance(y_values, str):
+            for datapoint in datapoints:
+                tmp = datapoint.copy()
+                # TODO What about inferred Y (y_values=None)?
+                tmp[REVISION_FIELD] = f"{revision}::{filename}"
```

Comment:
Not sure how to deal with this, but having this long strings as `rev` causes the Vega legend unusable:

<img width="549" alt="Captura de Pantalla 2022-05-09 a las 22 09 37" src="https://user-images.githubusercontent.com/12677733/167489451-3bf2c14f-1c3d-4efd-afe4-926c07efa94c.png">

Reply:
Apart from reducing font size (not very useful), I don't see which property could be adjusted:

https://vega.github.io/vega-lite-v1/docs/legend.html

Reply:
Alternative approach is to make `dvc plots show` edit datapoint inside renderer (just before HTML render, to not mess up with `vscode`) and remove common prefix.